zerver/views/messages.py

```diff
@@ -289,6 +289,30 @@ def by_pm_with(self, query, operand, maybe_negate):
                             column("recipient_id") == narrow_recipient.id))
             return query.where(maybe_negate(cond))
 
+    def by_huddle_with(self, query, operand, maybe_negate):
+        # type: (Query, str, ConditionTransform) -> Query
+        try:
+            narrow_profile = get_user_profile_by_email(operand)
+        except UserProfile.DoesNotExist:
+            raise BadNarrowOperator('unknown user ' + operand)
+
+        self_recipient_ids = [recipient_tuple['recipient_id']
+                              for recipient_tuple
+                              in Subscription.objects.filter(
+                              user_profile=self.user_profile,
+                              recipient__type=Recipient.HUDDLE
+                              ).values("recipient_id")]
+        narrow_recipient_ids = [recipient_tuple['recipient_id']
+                                for recipient_tuple
+                                in Subscription.objects.filter(
+                                user_profile=narrow_profile,
+                                recipient__type=Recipient.HUDDLE
+                                ).values("recipient_id")]
```

Comment:
I think you can use the `flat=True` option to `values()` to simplify this a bit.